akka-docs/src/main/paradox/scala/guide/tutorial_3.md

```diff
@@ -1,253 +1,180 @@
+This means that, for Akka messages:
 
-## Device Group
+ * If `M1` is delivered it must be delivered before `M2` and `M3`.
+ * If `M2` is delivered it must be delivered before `M3`.
+ * If `M4` is delivered it must be delivered before `M5` and `M6`.
+ * If `M5` is delivered it must be delivered before `M6`.
+ * `A2` can see messages from `A1` interleaved with messages from `A3`.
+ * Since there is no guaranteed delivery, any of the messages may be dropped, i.e. not arrive at `A2`.
 
-We are done with the registration support at the device level, now we have to implement it at the group level. A group
-has more work to do when it comes to registrations. It must either forward the request to an existing child, or it
-should create one. To be able to look up child actors by their device IDs we will use a @scala[`Map[String, ActorRef]`] @java[`Map<String, ActorRef>`].
+For the full details on delivery guarantees please refer to the [reference page](http://doc.akka.io/docs/akka/current/general/message-delivery-reliability.html).
 
-We also want to keep the original sender of the request so that our device actor can reply directly. This is possible
-by using `forward` instead of the @scala[`!`] @java[`tell`] operator. The only difference between the two is that `forward` keeps the original
-sender while @scala[`!`] @java[`tell`] always sets the sender to be the current actor. Just like with our device actor, we ensure that we don't
-respond to wrong group IDs:
+Reviewers: I wasn't sure if the "This means that" list of bullets *is* the ordering that Akka provides? I've edited to make it read that way. And, I think a diagram here would be really helpful, as well as a summary of the benefits (again).
 
-Scala
-:   @@snip [DeviceGroup.scala]($code$/scala/tutorial_3/DeviceGroup.scala) { #device-group-register }
+## Adding flexibility to device messages
 
-Java
-:   @@snip [DeviceGroup.java]($code$/java/jdocs/tutorial_3/DeviceGroup.java) { #device-group-register }
+Our first query protocol was correct, but did not take into account distributed application execution. If we want to implement resends in the actor that queries a device actor (because of timed out requests), or if we want to query multiple actors, we need to be able to correlate requests and responses. Hence, we add one more field to our messages, so that an ID can be provided by the requester.
 
-Just as we did with the device, we test this new functionality. We also test that the actors returned for the two
-different IDs are actually different, and we also attempt to record a temperature reading for each of the devices
-to see if the actors are responding.
+You can now add the following message definitions to your `IotApp` source file:
 
 Scala
-:   @@snip [DeviceGroupSpec.scala]($code$/scala/tutorial_3/DeviceGroupSpec.scala) { #device-group-test-registration }
+:   @@snip [DeviceInProgress.scala]($code$/scala/tutorial_2/DeviceInProgress.scala) { #read-protocol-2 }
```

Comment:
`tutorial_3.md` now refers to snippets in `tutorial_2/...` - let's move the example code, too.